core/src/bundle/usBundleUtils.cpp

```diff
@@ -31,151 +30,103 @@
 #define _GNU_SOURCE
 #endif
 #include <dlfcn.h>
+#if defined(__APPLE__)
+#include <sys/param.h>
+#include <mach-o/dyld.h>
+#endif
+#include <unistd.h>
+#include <limits.h>
 #elif _WIN32
 #include <windows.h>
-#endif
 
+#define RTLD_LAZY 0 // unused
 
-namespace us {
-
-namespace {
-#ifdef US_BUILD_SHARED_LIBS
-  const bool sharedLibMode = true;
-#else
-  const bool sharedLibMode = false;
-#endif
+const char* dlerror(void)
+{
+  static std::string errStr;
+  errStr = us::GetLastErrorStr();
+  return errStr.c_str();
 }
 
-namespace BundleUtils
+void* dlopen(const char * path, int mode)
 {
+  (void)mode; // ignored
+  return reinterpret_cast<void*>(path == nullptr ? GetModuleHandle(nullptr) : LoadLibrary(path));
+}
 
-#ifdef __GNUC__
-
-std::string GetLibraryPath_impl(void* symbol)
+void* dlsym(void *handle, const char *symbol)
 {
-  Dl_info info;
-  if (dladdr(symbol, &info))
-  {
-    return info.dli_fname;
-  }
-  else
-  {
-    US_DEBUG << "GetLibraryPath_impl() failed for address " << symbol;
-  }
-  return "";
+  return reinterpret_cast<void*>(GetProcAddress(reinterpret_cast<HMODULE>(handle), symbol));
 }
 
-void* GetSymbol_impl(const std::string& bundleName, const std::string& libLocation, const char* symbol)
-{
-  // Clear the last error message
-  dlerror();
+#endif
 
-  void* selfHandle = nullptr;
-  if (!sharedLibMode || bundleName == "main")
-  {
-    // Get the handle of the executable
-    selfHandle = dlopen(0, RTLD_LAZY);
-  }
-  else
-  {
-    selfHandle = dlopen(libLocation.c_str(), RTLD_LAZY);
-  }
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 1024
+#endif
 
-  if (selfHandle)
-  {
-    void* addr = dlsym(selfHandle, symbol);
-    if (!addr)
-    {
-      const char* dlerrorMsg = dlerror();
-      if (dlerrorMsg)
-      {
-        US_DEBUG << "GetSymbol_impl() failed: " << dlerrorMsg;
-      }
-    }
-    dlclose(selfHandle);
-    return addr;
-  }
-  else
-  {
-    US_DEBUG << "GetSymbol_impl() dlopen() failed: " << dlerror();
-  }
-  return nullptr;
-}
+namespace us {
 
-#elif _WIN32
 
-std::string GetLibraryPath_impl(void *symbol)
+namespace BundleUtils
 {
-  HMODULE handle = nullptr;
-  BOOL handleError = GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
-                                       static_cast<LPCTSTR>(symbol), &handle);
-  if (!handleError)
-  {
-    // Test
-    US_DEBUG << "GetLibraryPath_impl():GetBundleHandle() " << GetLastErrorStr();
-    return "";
-  }
-
-  char bundlePath[512];
-  if (GetModuleFileName(handle, bundlePath, 512))
-  {
-    return bundlePath;
-  }
-
-  US_DEBUG << "GetLibraryPath_impl():GetBundleFileName() " << GetLastErrorStr();
-  return "";
+  
+void* GetExecutableHandle()
+{
+  return dlopen(0, RTLD_LAZY);;
 }
 
-void* GetSymbol_impl(const std::string& bundleName, const std::string& libLocation, const char* symbol)
+std::string GetExecutablePath()
```

Comment:
That is because the executable is not linked with `-rdynamic`, as mentioned in https://github.com/CppMicroServices/CppMicroServices/pull/90#issuecomment-210188430. See also http://coliru.stacked-crooked.com/a/e68dcb2883c24514. I am okay with the  current implementation though.